src/main/java/health/ere/ps/service/dgc/DigitalGreenCertificateService.java

```diff
@@ -133,24 +138,39 @@ private String standardize(String fn) {
                 .header("Authorization", "Bearer " + getToken())
                 .post(entity);
 
-        byte[] responseData;
-
-        try {
-            responseData = response.readEntity(InputStream.class).readAllBytes();
-            if (Response.Status.Family.familyOf(response.getStatus()) != Response.Status.Family.SUCCESSFUL) {
-                throw new RuntimeException(new String(responseData));
+        switch (response.getStatus()) {
+            case 200: {
+                try {
+                    return response.readEntity(InputStream.class).readAllBytes();
+                } catch (IOException ioe) {
+                    throw new DgcCertificateServiceException(100200, "Could not read response from certificate service");
```

Comment:
I think it is better to capsle the errors in DgcError. Otherwise the error-codes must be documented.

Why adding a magic number (10000) to the error codes?

Reply:
DgcError is the DTO which does not contain any logic.